### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Ritchie333**
Two bug fixes found testing latest branch on Debian 10 with Ultimate Doom IWAD

1. SDL sound can be noisy, clearing the buffer fixes this
2. After completing any first map on any episode, play went on to E1M2 regardless of episode.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,20 @@ prboom2/data/prboom.wad
...
+data/Makefile
+data/cmake_install.cmake
+prboom2/data/rdatawad
+glboom-plus
+prboom-plus
+prboom-plus-game-server
+prboom-plus.wad
+src/CMakeFiles
+src/Makefile
+src/cmake_install.cmake
```
Comment (alexey-lysiuk) [inline .gitignore:23]: There is no need to add all this. Please use designated build directories named `build*` instead. While doing in-tree build isn't prohibited, it's not recommended either.

File: prboom2/src/g_game.c [hunk 2]
```diff
@@ -1586,6 +1586,12 @@ void G_DoCompleted (void)
 		  goto frommapinfo;	// skip past the default setup.
 	  }
   }
+  else
```
Comment (fabiangreffrath) [inline prboom2/src/g_game.c:1589]: Wasn't this already fixed by https://github.com/coelckers/prboom-plus/commit/dfe091e35a96e014d3f77c23b198f4a5004c6004 ?

Comment (Ritchie333) [reply prboom2/src/g_game.c:1589]: Looks like it given the date (6 Sep 2019) - the fix I did was some time before that but I only just thought to review changes now.

File: .gitignore (hunk 1 above)
Comment (Ritchie333) [reply .gitignore:23]: Should the documentation not say something about this? The INSTALL file still talks about using GNU autoconf.

File: prboom2/src/g_game.c (hunk 2 above)
Comment (fabiangreffrath) [reply prboom2/src/g_game.c:1589]: When you said "latest branch" I thought you meant that.

File: .gitignore (hunk 1 above)
Comment (alexey-lysiuk) [reply .gitignore:23]: If you can update it, please do ;)

**Comment by fabiangreffrath** [comment]
@Ritchie333 could you possibly post the sound buffer clearing fix as a separate PR, please?

**Comment by fabiangreffrath** [comment]
> @Ritchie333 could you possibly post the sound buffer clearing fix as a separate PR, please?

Commited, thanks!

**Pull request closed without merging**